walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,26 +27,29 @@
   public static final int MIN_INCREMENT = 10;
 
   /** 
-  * Returns a Collection of {@code TimeRange} objects that span any available meeting times for all people in the MeetingRequest request.
+  * Returns a Collection of {@link TimeRange} objects that 
+  * span any available meeting times for all people in the MeetingRequest request.
   * For example, if 4pm - 8pm is free for a meeting, 
-  * the Time Range object will represent that entire chunk of time rather than 4pm - 5pm, even if the duration of meeting is 1 hour. 
+  * the Time Range object will represent that 
+  * entire chunk of time rather than 4pm - 5pm, even if the duration of meeting is 1 hour. 
   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    long mtngDuration = request.getDuration();
-    Collection<String> attendees = request.getAttendees();
-    Collection<Event> allEvents = events;
-
-    // Mandatory events = events that must be attended by an attended in the Meeting Request
-    HashMap<Integer, Integer> mandatoryEvents = makeMandatoryEventMap(allEvents, attendees);
-    Collection<TimeRange> meetingTimes = findMeetingTimes(mtngDuration, mandatoryEvents);
+    // Mandatory events are events that must be attended by an attended in the {@link Meeting Request}
+    HashMap<Integer, Integer> mandatoryEvents = makeMandatoryEventMap(events, request);
+    Collection<TimeRange> meetingTimes = findMeetingTimes(request, mandatoryEvents);
     return meetingTimes;
   }
 
   /** 
-  * Creates a hash map <event start time, event end time> that ONLY contains meetings where 1 or more people from attendees must attend. 
+  * Creates a hash map <event start time, event end time> that ONLY contains meetings 
+  * where 1 or more people from attendees must attend. 
+  * If there are two meetings with the same start time,
+  * only the one with a later end time is saved.
   **/
-  public HashMap<Integer, Integer> makeMandatoryEventMap (Collection<Event> allEvents, Collection<String> attendees) {
+  public HashMap<Integer, Integer> makeMandatoryEventMap (Collection<Event> events, MeetingRequest request) {
     HashMap<Integer, Integer> mandatoryEvents = new HashMap<Integer, Integer>();
+    Collection<String> attendees = request.getAttendees();
+    ArrayList<Event> allEvents = new ArrayList<>(events);
```

Comment:
Because you're not modifying anything in `events`, no need to create a copy of it, so I think this should just be removed and use `events` in the `for` loop.